daemon/containerd/image_save_test.go

```diff
@@ -0,0 +1,103 @@
+package containerd
+
+import (
+	"context"
+	"io"
+	"path/filepath"
+	"testing"
+
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/platforms"
+	"github.com/docker/docker/errdefs"
+	"github.com/docker/docker/internal/testutils/specialimage"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"gotest.tools/v3/assert"
+	is "gotest.tools/v3/assert/cmp"
+)
+
+func TestImageMultiplatformSaveShallowWithNative(t *testing.T) {
+	ctx := namespaces.WithNamespace(context.TODO(), "testing-"+t.Name())
+
+	contentDir := t.TempDir()
+	store := &blobsDirContentStore{blobs: filepath.Join(contentDir, "blobs/sha256")}
+
+	native := platforms.Platform{
+		OS:           "linux",
+		Architecture: "amd64",
+	}
```

Comment:
Considering if we should have a really fictive platform for `native` (e.g. `cpm/z80`); I know we're mocking things, but to reduce risk of a test falling through an matching _actual_ native platforms (which can differ where you're running).

(also not a blocker, mostly thinking out loud here)q